mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -117,7 +121,8 @@ public MessageSearches() {
                         input.getMailboxId(),
                         input.getUid());
                 }
-            }).toSet();
+            })
+            .toSet();
```

Comment:
Set doesn't keep ordering, only SortedSet does. How can you know that toSet will keep ordering ?